### ISSUE TITLE
Used Deprecated Functions not available on page-load

### ISSUE THREAD

**Issue by StachuDotNet** [darklang/dark#4481]
If a canvas contains a deprecated function, we intend for users to be able to use that function, even being deprecated.

This works generally, but not on first page load.

Specifically, it seems that the `usedFns` object passed into the `props` used by `calculateAllowedFunctionsList` is empty, until `SetUserFunctions` or `LoadPackagesAPICallback` messages come through.

**Comment by StachuDotNet** [comment]
The logic of updating `functions` in the model, via `Functions.update`, happens in several places: init, on `SetUserFunctions` and `LoadPackagesAPICallback` fns, and now with #4482, also when updating in-progress feature settings.

This repeated logic makes me think we might be well off with something like:
- define a new `ReevaluateFunctions` Msg to refresh the available fns
- when processing a `ReevaluateFunctions` Msg, 
  ```
    let props: Functions.props = {
      usedFns: m.usedFns,
      userFunctions: m.userFunctions,
      allowTuples: m.settings.contributingSettings.inProgressFeatures.allowTuples,
    }
    let m = {...m, functions: Functions.update(props, m.functions)}
    processAutocompleteMods(m, list{ACRegenerate})
  ```
- update the existing 3 (now 4, with #4482) places where we create the `props` to yield a `ReevaluateFunctions` rather than repeat that logic.
- yield `ReevaluateFunctions` after page-load, so that this logic is triggered

I'm not sure if these changes are sound, though - for the `LoadPackagesAPICallback` Msg handling, there's a `let m = Introspect.refreshUsages(m, Map.keys(m.handlers))` line there that I'm not sure is appropriate to include in a general `ReevaluateFunctions` case.

Maybe this is the wrong direction, and we shouldn't try to consolidate updating functions?

It's probably worth a short call to discuss; I've been having some difficulty with the Intent/Msg/Cmd dance, in any case.

**Comment by StachuDotNet** [comment]
This has been addressed in #4482 by ensuring we refresh the fns appropriately upon `InitialLoadAPICallback`.
Specifically, we updated `SetToplevels` which is triggered by `InitialLoadAPICallback`.
